src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -350,14 +364,13 @@ internal static async Task<ExceptionalConditions> ProcessSarifLogAsync(Stream st
             }
             catch (JsonReaderException)
             {
+                RaiseLogProcessed(ExceptionalConditions.InvalidJson);
             }
 
             if (sarifLog != null)
             {
                 await ProcessSarifLogAsync(sarifLog, logFilePath: logId, showMessageOnNoResults: showMessageOnNoResults, cleanErrors: cleanErrors, openInEditor: openInEditor);
             }
-
-            return ExceptionalConditionsCalculator.Calculate(sarifLog);
```

Comment:
>ExceptionalConditionsCalculator [](start = 19, length = 31)

Moved into `ProcessSarifLogAsync(SarifLog)` overload. That's what makes it work with the file-based APIs, because everything ultimately calls into the `SarifLog` overload.